lib/rule-tester/flat-rule-tester.js

```diff
@@ -0,0 +1,1061 @@
+                     * parsers will be wrapped multiple times.
+                     */
+                    parsers: {
+                        ...defaultConfig[0].plugins["@"].parsers
+                    },
+
+                    /*
+                     * The rules key on the default plugin is a proxy to lazy-load
+                     * just the rules that are needed. So, don't create a new object
+                     * here, just use the default one to keep that performance
+                     * enhancement.
+                     */
+                    rules: defaultConfig[0].plugins["@"].rules
+                },
+                "rule-to-test": {
+                    rules: {
+                        [ruleName]: Object.assign({}, rule, {
+
+                            // Create a wrapper rule that freezes the `context` properties.
+                            create(context) {
+                                freezeDeeply(context.options);
+                                freezeDeeply(context.settings);
+                                freezeDeeply(context.parserOptions);
+
+                                // freezeDeeply(context.languageOptions);
+
+                                return (typeof rule === "function" ? rule : rule.create)(context);
+                            }
+                        })
+                    }
+                }
+            },
+            languageOptions: {
+                ...defaultConfig[0].languageOptions
+            }
+        };
+
+        // wrap default parsers
+        for (const [parserName, parser] of Object.entries(baseConfig.plugins["@"].parsers)) {
+            baseConfig.plugins["@"].parsers[parserName] = wrapParser(parser);
+        }
+
+        /**
+         * Run the rule for the given item
+         * @param {string|Object} item Item to run the rule against
+         * @throws {Error} If an invalid schema.
+         * @returns {Object} Eslint run result
+         * @private
+         */
+        function runRuleForItem(item) {
+            const configs = new FlatConfigArray(testerConfig, { baseConfig });
+            let code, filename, output, beforeAST, afterAST;
+
+            if (typeof item === "string") {
+                code = item;
+            } else {
+                code = item.code;
+
+                /*
+                 * Assumes everything on the item is a config except for the
+                 * parameters used by this tester
+                 */
+                const itemConfig = { ...item };
+
+                for (const parameter of RuleTesterParameters) {
+                    delete itemConfig[parameter];
+                }
+
+                // wrap any parsers
+                if (itemConfig.languageOptions && itemConfig.languageOptions.parser) {
+
+                    const parser = itemConfig.languageOptions.parser;
+
+                    if (parser && typeof parser !== "object") {
```

Comment:
```suggestion
                    if (typeof parser !== "object") {
```

At this point `parser` is certainly truthy.

Reply:
I'd prefer to leave this in for clarity (I always check for truthiness before checking if `typeof` is an object just to be safe.)